sort/mergesort.go

```diff
@@ -45,3 +47,13 @@ func Mergesort(items []int) []int {
 	return merge(a, b)
 
 }
+
+func MergesortIter(items []int) []int {
```

Comment:
`MergeIter` not `MergesortIter`. Sort is in the package name.
https://go.dev/doc/effective_go#package-names